src/Functions/fromReadableSize.cpp

```diff
@@ -0,0 +1,70 @@
+#include <base/types.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/fromReadable.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int CANNOT_PARSE_TEXT;
+}
+
+namespace
+{
+
+// ISO/IEC 80000-13 binary units
+const std::unordered_map<std::string_view, Float64> scale_factors =
+{
+    {"b", 1.0},
+    {"kib", 1024.0},
+    {"mib", 1024.0 * 1024.0},
+    {"gib", 1024.0 * 1024.0 * 1024.0},
+    {"tib", 1024.0 * 1024.0 * 1024.0 * 1024.0},
+    {"pib", 1024.0 * 1024.0 * 1024.0 * 1024.0 * 1024.0},
+    {"eib", 1024.0 * 1024.0 * 1024.0 * 1024.0 * 1024.0 * 1024.0},
+};
+
+struct Impl
+{
+    static constexpr auto name = "fromReadableSize";
+
+    static Float64 getScaleFactorForUnit(const String & unit)  // Assumes the unit is already in lowercase
+    {
+        auto iter = scale_factors.find(unit);
+        if (iter == scale_factors.end())
+        {
+            throw Exception(
+                ErrorCodes::CANNOT_PARSE_TEXT,
+                "Invalid expression for function {} - Unknown readable size unit (\"{}\")",
+                name,
+                unit
+            );
+        }
+        return iter->second;
+    }
+
+};
+}
+
+REGISTER_FUNCTION(FromReadableSize)
+{
+    factory.registerFunction<FunctionFromReadable<Impl>>(FunctionDocumentation
+        {
+            .description=R"(
```

Comment:
Having in-source docs like here is required for new functions (otherwise a test in CI checking specifically this will fail). So far so good.

If you look at Common/FunctionDocumentation.h, you see that there are more fields:

```cpp
Description description;
Syntax syntax = {};
Arguments arguments {};
ReturnedValue returned_value {};
Examples examples {};
Categories categories {};
```

It would be great to populate these as well. The markdown and in-source docs are ideally kept in-sync. In future, we plan to generate the former from the latter. Also, designated initialization syntax is (IMHO) a bit ugly for this. Feel free to create a standalone `FunctionDocumentation` object:

```cpp
FunctionDocumentation::Description description = "...";
FunctionDocumentation::Syntax syntax = "...";
[...]

factory.registerFunction<FunctionFromReadable<Impl>>({description, syntax, arguments, returned_value, examples, categories});
```

^^ Maybe I did not get the syntax right but I hope got the point :-)

Reply:
Point gotten, thanks for the pointer! I've added all fields in dedicated `FunctionDocumentation` instances for each one of the types as although similar each function has somewhat different examples and descriptions. Added a question [here](https://github.com/ClickHouse/ClickHouse/pull/64386/files#r1617103271) about it